src/components/ProgressCircular/progressRing.js

```diff
@@ -1,10 +1,11 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 
 export default function ProgressRing({ percent }) {
-    const isLongArc = percent > 50 ? 1 : 0;
-    const arcX = Math.cos((2 * Math.PI * percent) / 100);
-    const arcY = Math.sin((2 * Math.PI * percent) / 100);
+    const [percentValue, setPercentValue] = useState(0);
+    useEffect(() => {
+        setPercentValue(percent);
+    }, [percent]);
```

Comment:
here if you only call setPercentValue when percent change then you can use percent directly, so there is not need to use useState and useEffect